pkg/agent/utils.go

```diff
@@ -371,21 +371,23 @@ func GetOrderedKubeletConfigFlagString(k *datamodel.KubernetesConfig, cs *datamo
 	return buf.String()
 }
 
-// IsDynamicKubeletEnabled get if dynamic kubelet is supported in AKS and toggle is on
-func IsDynamicKubeletEnabled(cs *datamodel.ContainerService, dynamicKubeletToggleEnabled bool) bool {
+// IsKubeletConfigFileEnabled get if dynamic kubelet is supported in AKS and toggle is on
+func IsKubeletConfigFileEnabled(cs *datamodel.ContainerService, profile *datamodel.AgentPoolProfile, kubeletConfigFileToggleEnabled bool) bool {
 	// TODO(bowa) remove toggle when backfill
-	return dynamicKubeletToggleEnabled && cs.Properties.OrchestratorProfile.IsKubernetes() && IsKubernetesVersionGe(cs.Properties.OrchestratorProfile.OrchestratorVersion, "1.14.0")
+	// If customKubeletConfig or customLinuxOSConfig is used (API20201101 and later), use kubelet config file
+	return profile.CustomKubeletConfig != nil || profile.CustomLinuxOSConfig != nil ||
+		kubeletConfigFileToggleEnabled && cs.Properties.OrchestratorProfile.IsKubernetes() && IsKubernetesVersionGe(cs.Properties.OrchestratorProfile.OrchestratorVersion, "1.14.0")
```

Comment:
nit: can we add a round bracket to make it clearer ?
 (kubeletConfigFileToggleEnabled && cs.Properties.OrchestratorProfile.IsKubernetes() && IsKubernetesVersionGe(cs.Properties.OrchestratorProfile.OrchestratorVersion, "1.14.0"))

Reply:
fixed